airflow/providers/google/cloud/log/stackdriver_task_handler.py

```diff
@@ -99,7 +99,7 @@ def __init__(
         super().__init__()
         self.gcp_key_path: str | None = gcp_key_path
         self.scopes: Collection[str] | None = scopes
-        self.name: str = name
+        self.gcp_log_name: str = name
```

Comment:
I'm not sure that I fully understand the essence of this change - is the use of `self.name` causing a current bug due to its override, or is it solely intended for differentiation to prevent potential future bugs?

At least for the latter case - it's better to make `self.name` a deprecated property as it breaks backward compatibility (see example [here](https://github.com/shahar1/airflow/blob/d4d4ca38cc3a65f3d61bc6baa6a5eef4de740284/airflow/providers/google/cloud/operators/vertex_ai/custom_job.py#L1360)).

Reply:
Sorry, maybe I forgot to add some context ^^
So, Airflow relies on the [`logging.config.dictConfig`](https://docs.python.org/3/library/logging.config.html) method to [setup the logging stack](https://github.com/apache/airflow/blob/a58441ca1b263cae61a5bb653e6839f0dd29b08e/airflow/logging_config.py#L69).
However, during this setup, it iterates through the handlers and [explicitly sets their name](https://github.com/python/cpython/blob/2a4cbf17af19a01d942f9579342f77c39fbd23c4/Lib/logging/config.py#L578):
```py
                for name in sorted(handlers):
                    try:
                        handler = self.configure_handler(handlers[name])
                        handler.name = name
                        handlers[name] = handler
```

So, before this fix:

1. Setup the remote logging through the environment variables `AIRFLOW__LOGGING__REMOTE_LOGGING="true"` and `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER="stackdriver://host/path"`
2. Airflow instanciates a `StackdriverTaskHandler` with the name of `"path"`
3. **BUT** the `dictConfig` call overrides the name of the handler with the key of the handlers configuration (i.e. [`task`](https://github.com/apache/airflow/blob/a58441ca1b263cae61a5bb653e6839f0dd29b08e/airflow/config_templates/airflow_local_settings.py#L350)).
4. Hence, the next calls to the `emit` method of the handler will generate logs to the wrong destination (`task` instead of `path`).

Changing the field, from `name` to `gcp_log_name` prevents the overriding from the `dictConfig`. For me, it is a bug fix (and not a change to prevent potential future bugs), but as you noted, it breaks the backward compatibility. Should we still pass through a deprecation period or should we update it ASAP ?

Reply:
Thanks for the context :)

@eladkal Given that the bug fix would be breaking due to the apparent reservation of the attribute `name`, what do we do in this case in terms of deprecation?

Reply:
Nope, but it would be nice adding an indicative exception in case someone passes the `name` param

Reply:
I don't see any "good" use case where someone would want the current behavior instead of the fix.
@shahar1 I don't change the constructor arguments (you can still get an instance using `StackdriverTaskHandler(name="foo", ...)`) but I change the meaning of the instance attribute `name` (up to now it was supposed to be the name of the GCP log, after this fix it will be the name of the Python handler).
I may add some logs on access (read or write) to the `name` attribute, but I think it may be hard to implement cleanly and will be too verbose (as it is a genuine behavior of the `dictConfig` method).